integration-tests/src/main/java/io/quarkus/it/jsch/JSchResource.java

```diff
@@ -8,14 +9,25 @@
 import jakarta.ws.rs.core.Response;
 
 import com.jcraft.jsch.JSch;
+import com.jcraft.jsch.JSchException;
 import com.jcraft.jsch.KeyPair;
 import com.jcraft.jsch.Session;
 
+import io.quarkus.jsch.JSchSession;
+import io.quarkus.jsch.JSchSessions;
+
 @Path("/jsch")
+@RequestScoped
 public class JSchResource {
 
+    @JSchSession("doudou")
```

Comment:
I think using `io.smallrye.common.annotation.Identifier` would be better. Plus, you shouldn't need to use `@JSchSession` given that the injection type expected is already a Session

Reply:
Yes I can do it with `@Identifier`.

But I want to be consistent with the future works: create a session by a simple annotation is cool, but it will be even better to create a channel.

I have already thought about it:

```java
package io.quarkus.jsch;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target({ ElementType.FIELD })
public @interface JschChannel {

    /**
     * The channel type.
     */
    String value();

    /**
     * The session name.
     */
    String session() default JSchSession.DEFAULT_SESSION_NAME;

}
```

And i havent find a way to do it with just a simple naming annotation.
I wanted to be consistent with all this extension's annotations. But yes, we can use `@Identifier` for sessions and for the channels we can see it later and use a custom one.

I'll let you choose the solution that seems the cleanest to you.

Reply:
The injections types for this future annotation are:

* com.jcraft.jsch.Channel
* com.jcraft.jsch.ChannelSftp
* com.jcraft.jsch.ChannelShell
* com.jcraft.jsch.ChannelExec